Buildings/Controls/OBC/ASHRAE/G36/Atomic/TrimRespondLogic.mo

```diff
@@ -0,0 +1,313 @@
+  CDL.Continuous.Sources.Constant resAmoCon(k=resAmo) "Respond amount constant"
+    annotation (Placement(transformation(extent={{-80,-160},{-60,-140}})));
+  CDL.Continuous.Product pro
+    "Products of net requests and respond amount value"
+    annotation (Placement(transformation(extent={{-20,-140},{0,-120}})));
+  CDL.Discrete.UnitDelay uniDel(
+    samplePeriod=timSte,
+    y_start=iniSet,
+    startTime=delTim + timSte) "Output the input signal with a unit delay"
+    annotation (Placement(transformation(extent={{-80,50},{-60,70}})));
+  CDL.Logical.Switch swi "Switch between initial setpoint and reseted setpoint"
+    annotation (Placement(transformation(extent={{160,140},{180,120}})));
+  CDL.Logical.Switch swi1
+    "Before instant (device ON + delTim + timSte), the setpoint should not be trimmed"
+    annotation (Placement(transformation(extent={{80,-40},{100,-20}})));
+  CDL.Logical.Switch swi2
+    "Reinitialize setpoint to initial setting when device become OFF"
+    annotation (Placement(transformation(extent={{120,50},{140,70}})));
+  CDL.Discrete.Sampler sampler(
+    samplePeriod=timSte,
+    startTime=delTim + timSte) "Sample number of requests"
+    annotation (Placement(transformation(extent={{-140,-100},{-120,-80}})));
+
+protected
+  CDL.Continuous.Sources.Constant iniSetCon(k=iniSet) "Initial setpoint"
+    annotation (Placement(transformation(extent={{-80,140},{-60,160}})));
+  CDL.Continuous.Sources.Constant numIgnReqCon(k=numIgnReq)
+    "Number of ignored requests"
+    annotation (Placement(transformation(extent={{-140,-60},{-120,-40}})));
+  CDL.Continuous.Sources.Constant triAmoCon(k=triAmo) "Trim amount constant"
+    annotation (Placement(transformation(extent={{-80,-120},{-60,-100}})));
+  CDL.Continuous.Sources.Constant maxResCon(k=maxRes)
+    "Maximum response per time interval"
+    annotation (Placement(transformation(extent={{-20,-180},{0,-160}})));
+  CDL.Continuous.Sources.Constant maxSetCon(k=maxSet)
+    "Maximum setpoint constant"
+    annotation (Placement(transformation(extent={{-20,20},{0,40}})));
+  CDL.Continuous.Sources.Constant zerTri(k=0)
+    "Zero reset amount during time range from (device ON) to (device ON + delTim + timSet)"
+    annotation (Placement(transformation(extent={{20,-60},{40,-40}})));
+  CDL.Conversions.IntegerToReal intToRea "Convert integer input to real output"
+    annotation (Placement(transformation(extent={{-180,-100},{-160,-80}})));
+  CDL.Continuous.Add difReqIgnReq(k1=-1)
+    "Difference between ignored request number and the real request number"
+    annotation (Placement(transformation(extent={{-80,-80},{-60,-60}})));
+  CDL.Continuous.Add add1
+    "Increase setpoint by amount of value defined from reset logic"
+    annotation (Placement(transformation(extent={{-20,50},{0,70}})));
+  CDL.Continuous.Add add2 "Net reset value"
+    annotation (Placement(transformation(extent={{80,-140},{100,-120}})));
+  CDL.Continuous.Min min
```

Comment:
Is this block name allowed? There is also a ```max``` block in line 98. Those strings might cause ambiguity since they also represent a mathematical operation. I don't know how Modelica handles this.

Reply:
@JayHuLBL  @milicag :  It is better not to call the instance ``min`` or ``max``. Modelica allows it, but when we convert it to other languages, this could cause problems.